rfcs/selections.md

```diff
@@ -0,0 +1,108 @@
+---
+Stage: Draft
+Start Date: 30-09-2022
+Release Date: unreleased
+---
+
+# Improvements in how to represent and handle document selections
+
+## Summary
+
+This RFC includes a proposal in how we can represent ModelSelections in a more versatile way. It introduces some different types of selections and abstractions built on top of the currently used ModelSelection inspired by the selection representations in [ProseMirror](https://prosemirror.net/).
+
+Additionally this RFC also includes some proprosals on how we handle and display these selections in the editor.
+
+## Motivation
+
+As the editor API evolves, documents have the ability to contain specific elements which do not only represent text. With the introduction of inline components, interactive elements such as buttons and image can be inserted in a document. Currently, the editor does not yet include functionality on how to select these non-text elements in an intuitive way.
+
+This RFC proposes a design on how we can introduce abstraction on top of the current ModelSelection design.
+
+Additionally it presents some ways in how we can handle these abstractions and how we can represent them in the document user view.
+
+## Detailed design
+
+### Representation of selections
+
+This section introduces two selection implementations
+
+- A `BasicSelection`, which represents the `ModelSelections` we are used to. These are used to represents selections of text and the whole document. In addition to the selection ranges, this selection can also contain marks and have a direction.
+
+- A `ComponentSelection`, which represents a selection of an Inline Component. These selections are triggered by e.g. clicking on such a component. Aside from the selection ranges, this selection type also keeps track of the selected component. This type of selection is only used when selecting a component and nothing else.
+
+Both implementations build upon the `ModelSelection` abstract class, which keeps track of the ranges contained in the selection.
+
+```typescript
+abstract class ModelSelection {
+  private _ranges: ModelRange[];
+
+  constructor(ranges: ModelRange[] = []) {
+    this._ranges = ranges;
```

Comment:
I'd prefer to see this without implementation, it doesn't really belong in an RFC
It can be relevant to describe plans for a particular implementation to demonstrate the feasibility of an interface, but I don't really see the relevance here